### PR TITLE
Add "Insert Checklist" menu item

### LINKED ISSUE #1138: Add menu item to insert checklist
Format ▸ Insert Checklist <kbd>⌘⌥C</kbd>

- Should insert as well as remove checkboxes to the current line.
- If multiple lines are selected, it should add/remove checkboxes to all of them.

### PR DESCRIPTION
**Pull request by mirka**
Closes #1138 

## Single
insert-checklist-single

## Multiple
insert-checklist-multiple

### CODE REVIEW

File: lib/editor/insert-or-remove-checkboxes.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  const text = contentState.getBlockForKey(blockKey).getText();
+  taskRegex.lastIndex = 0; // Reset the search
+  const matches = taskRegex.exec(text);
+  let newContentState;
+
+  if (matches) {
+    // Remove task prefix
+    const leadingWhitespace = get(matches, '1', '');
+    const taskPrefix = matches[2];
+    const spacesToTrim = matches[3].match(/^\s*/)[0].length;
```
Comment (roundhill) [inline lib/editor/insert-or-remove-checkboxes.js]: Are we safe from out of index exceptions here?

**Comment by roundhill** [comment]
Not sure if this is the right PR to report this, but I noticed that an extra space is hanging around if you press enter on an empty checklist item and it gets removed:

space-character

File: lib/editor/insert-or-remove-checkboxes.js (hunk 1 above)
Comment (mirka) [reply lib/editor/insert-or-remove-checkboxes.js]: Should be ok, and unit tests will fail spectacularly if the regex is changed in an unexpected way. I tweaked the regex a bit to ensure that the first capturing group is always a string though (https://github.com/Automattic/simplenote-electron/pull/1166/commits/0a738e6a49d1ae8d22ce0c549003d335e3e7fb63).

**Comment by mirka** [comment]
> I noticed that an extra space is hanging around if you press enter on an empty checklist item and it gets removed

Great catch. I've determined the cause and will fix in a separate PR. Thanks!

**Pull request merged**